examples/data_preprocessing/wiennerisches_diarium_historical_dataset/data_preprocessing_sterbelisten_wiener_diarum.ipynb

```diff
@@ -0,0 +1,512 @@
+    "#create an empty list and split the text at line break, creating an element for each sentence\n",
+    "\n",
+    "sterbeliste = []\n",
+    "sterbeliste = sterbelisten_strip_html2.split(\"\\n\\n\\n\")\n",
+    "#print(sterbeliste[0])"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "## 2.Create labeled sentences and a dictionary of place names"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "Open questions:\n",
+    "\n",
+    "This code works on the assumption that both \\<mark> and \\<\\mark> tags are present.\n",
+    "There are 14870 \\<mark> and 11646 \\<\\mark> tags. \n",
+    "One solution could be exluding the 3224 items and their samples from our list, or check them manually and use a cleaned html file. \n"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 14,
+   "metadata": {},
+   "outputs": [
+    {
+     "name": "stdout",
+     "output_type": "stream",
+     "text": [
+      "Number of <mark> tags: 14870\n",
+      "Number of </mark> tags: 11646\n",
+      "Number of unclosed tags: 3224\n"
+     ]
+    }
+   ],
+   "source": [
+    "print(f\"Number of <mark> tags: {len(re.findall('<mark>',sterbelisten_html))}\")\n",
+    "print(f\"Number of </mark> tags: {len(re.findall('</mark>',sterbelisten_html))}\")\n",
+    "print((f\"Number of unclosed tags: {len(re.findall('<mark>',sterbelisten_html))-len(re.findall('</mark>',sterbelisten_html))}\"))"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 11,
+   "metadata": {},
+   "outputs": [
+    {
+     "name": "stdout",
+     "output_type": "stream",
+     "text": [
+      "Length of the sentence: 13\n",
+      "Sample_tagging:[None, None, None, None, None, None, 0, 0, None, 1, 1, None, None, None, None, None, None, None]\n",
+      "Number of place names: 6\n",
+      "=======================================\n",
+      "Sentense and labeles: [('Christina', None), ('Kochin', None), ('ein', None), ('Wittib', None), ('im', None), ('Barbieris', 4), ('Hauß', 4), ('auf', None), ('der', None), ('Laimgrueben', 5), ('alt', None), ('67', None), ('Jahr', None)]\n",
+      "Cleaned Sentense: ['\\nDem', 'Peter', 'Frost', 'einem', 'Cammer', 'im', 'Greiseckeris', 'Hauß', 'im', 'Diener', 'Gäßl', 'sein', 'Kind', 'Frantz', 'alt', '6', 'viertl', 'Jahr']\n"
+     ]
+    }
+   ],
+   "source": [
+    "def get_location_id(location: List, locations: List) -> int:\n",
+    "    '''\n",
+    "    this function loops through the list of place names, \n",
+    "    if it doesn't find a given place name it adds the place name to the list\n",
+    "    and returns the new list \n",
+    "    \n",
+    "    input: the list or single word that is part of a place name\n",
+    "    output: list of place names \n",
+    "    '''\n",
+    "    for word in location:\n",
+    "        if word not in locations:\n",
+    "            locations[word]= len(locations)\n",
+    "            return locations[word]\n",
```

Comment:
you should also add `else` branch, i.e. what will the function return if `word in locations`?

Reply:
also: I would suggest changing the naming a bit because `location` and `locations` sound too similar

Reply:
also-2: it is better to convert this part of code into the separate function + write a test for it (let's talk about it)

Reply:
docstring: 
```
"""
[discription what function does]
Args:
    curr_location: 
    all_locations: 
Returns: 
"""
```